editor/scene_tree_dock.cpp

```diff
@@ -1307,6 +1307,13 @@ bool SceneTreeDock::_validate_no_foreign() {
 			return false;
 		}
 
+		// When edited_scene inherits from another one the root Node will be the parent Scene,
+		// we don't want to consider that Node a foreign one otherwise we would not be able to
+		// delete it
+		if (edited_scene->get_scene_inherited_state().is_valid() && edited_scene->get_path_to(E->get()) == NodePath(".")) {
```

Comment:
I think `edited_scene` is the root node already, so you could likely do `E->get() == edited_scene` instead of comparing NodePaths (not tested though, might be wrong).